.github/workflows/main.yml

```diff
@@ -61,13 +61,16 @@ jobs:
         run: sudo apt-get install -y libyaml-dev rpm
       - name: Install Ansible
         run: pip install --upgrade https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz
+      - name: Ensure Jinja version is old enough for Ansible 2.9
+        run: pip install --upgrade 'Jinja2<3.1'
+        if: matrix.ansible == 'v2.9.17'
```

Comment:
only later versions on 2.9 are compatible with Jinja 3.1+, and I kinda think it's up to the user to have a working environment